### PR TITLE
feat(modal): instant activation of modal on click rather than on mouseUp

### PR DESCRIPTION
**Pull request by kembreyfarquhar**
Existing code doesn't activate modal until user releases their mouse click. This causes the modal
dragging to be behind the currently active modal. This fix/feature immediately activates the modal
window on click, allowing it to drag in front of all other modals.

* Also adds a "Multiple" story to the Modal section of Storybook, demonstrating how the changes work

### CODE REVIEW

File: packages/core/components/Modal/Modal.tsx
Comment (ggdaltoso) [inline packages/core/components/Modal/Modal.tsx:253]:
```suggestion
    <Draggable 
         handle=".draggable" 
         defaultPosition={defaultPosition} 
         onMouseDown={id ? () => setActiveWindow(id) : undefined}
       >
```

💅

**Review by ggdaltoso — COMMENTED** [review]
Easy fix for a big improvement! 

Thanks for your contribution, @kembreyfarquhar !

I've just added a comment asking you to ident your changes. I'll wait for your reply but I will merge it if it looks fine to you. Although, everything seems good to me!

File: packages/core/components/Modal/Modal.tsx
Comment (kembreyfarquhar) [reply packages/core/components/Modal/Modal.tsx]: I applied the suggested indentation :)

**Comment by kembreyfarquhar** [comment]
Oh no, codesandbox build failed. I'm not sure why.. However I forgot to run `yarn commit` and used `git commit` instead. Would that cause an issue? @ggdaltoso

**Review by ggdaltoso — APPROVED** [review]

**Comment by ggdaltoso** [comment]
Very weird, @kembreyfarquhar...

But, after running all publish steps locally, I'm certain to say it's all good. 

We could investigate what happened with Codesandbox CI in another issue/PR. Ok?

**Comment by ggdaltoso** [comment]
@all-contributors please add @kembreyfarquhar for code and bug

**Pull request merged**
